### PR TITLE
[VR-11345] Fix transaction deadlock error

### PR DESCRIPTION
**Pull request by anandjakhaniya**
Jira Ticket: https://vertaai.atlassian.net/browse/VR-11345

### CODE REVIEW

**Comment by anandjakhaniya** [comment]
Used following guidelines: https://www.percona.com/community-blog/2018/09/24/minimize-mysql-deadlocks-3-steps/ **(Use a lock-avoiding design strategy)**

And Reference from mysql: https://dev.mysql.com/doc/refman/8.0/en/innodb-transaction-isolation-levels.html

File: backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CreateExperimentRunHandler.java [hunk 1]
```diff
@@ -241,8 +239,19 @@ private ExperimentRun getExperimentRunFromRequest(
...
-                      int count = query.execute();
-                      LOGGER.trace("ExperimentRun Inserted : " + (count > 0));
+
+                      try {
+                        int count = query.execute();
+                        LOGGER.trace("ExperimentRun Inserted : " + (count > 0));
+                      } catch (UnableToExecuteStatementException exception) {
+                        if (exception.getCause() instanceof MySQLTransactionRollbackException) {
+                          // take a brief pause before resubmitting its query/transaction
+                          Thread.sleep(100); // Time in ms
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CreateExperimentRunHandler.java]: 100ms is not brief by any stretch =)

**Review by conradoverta — APPROVED** [review]
We should place the retry logic in our jdbi wrapper as it will certainly happen again.

File: backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CreateExperimentRunHandler.java (hunk 1 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CreateExperimentRunHandler.java]: How much time is ideal here??

**Comment by anandjakhaniya** [comment]
> We should place the retry logic in our jdbi wrapper as it will certainly happen again.

Will add in follow up PR

File: backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CreateExperimentRunHandler.java (hunk 1 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CreateExperimentRunHandler.java]: Added time based on config setting

**Review by conradoverta — APPROVED** [review]

**Pull request merged**